testing/examples/WORKSPACE

```diff
@@ -63,6 +63,7 @@ container_pull(
 
 dockerfile_image(
     name = "basic_alpine_dockerfile",
+    cache = True,
```

Comment:
This cant really be tested properly without doing an e2e test. 
The test might do the following:
setup  - have a rule w cache = True in this `WORSKPACE` file (a different target, though)
1- Create an image from dockerfile w/o using bazel, check its  sha
2- Create an image using Bazel. Check its sha256 is the same as the one produced in step 1
As a base case when not enabled the test would need to:
setup  - have the exact same  w/o cache = True (reuse this `basic_alpine_dockerfile` target) 
3- build the image using bazel
4- check its sha is different from the one produced in step 1 and 2.

One key question I have is if we have all the outputs in this rule already to expose the sha so you can do the check you need to do.